src/register.rs

```diff
@@ -6,13 +6,143 @@ pub mod common {
     use bit_field::BitArray;
 
     pub const MODE: u16 = 0x0;
+
+    /// Register: GAR (Gateway IP Address Register) [R/W] [0x0001 – 0x0004] [0x00]
     pub const GATEWAY: u16 = 0x01;
+
+    /// Register: SUBR (Subnet Mask Register) [R/W] [0x0005 – 0x0008] [0x00]
     pub const SUBNET_MASK: u16 = 0x05;
+
+    /// Register: SHAR (Source Hardware Address Register) [R/W] [0x0009 – 0x000E] [0x00]
     pub const MAC: u16 = 0x09;
+
+    /// Register: SIPR (Source IP Address Register) [R/W] [0x000F – 0x0012] [0x00]
     pub const IP: u16 = 0x0F;
+
+    /// Register: INTLEVEL (Interrupt Low Level Timer Register) [R/W] [0x0013 – 0x0014] [0x0000]
+    pub const INTERRUPT_TIMER: u16 = 0x13;
+
+    /// Register: RTR (Retry Time-value Register) [R/W] [0x0019 – 0x001A] [0x07D0]
+    pub const RETRY_TIME: u16 = 0x19;
+
+    /// Register: RCR (Retry Count Register) [R/W] [0x001B] [0x08]
+    pub const RETRY_COUNT: u16 = 0x1B;
+
     pub const PHY_CONFIG: u16 = 0x2E;
     pub const VERSION: u16 = 0x39;
 
+    /// The common register Mode.
+    ///
+    /// It differs from the [`crate::Mode`] in one key aspect - we can also
+    /// send a reset value to the w5500, instead of only configuring the settings.
+    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    pub enum Mode {
+        Reset,
+        Mode(crate::Mode),
+    }
+
+    impl Mode {
+        pub fn to_register(self) -> [u8; 1] {
+            [self.to_u8()]
+        }
+
+        pub fn to_u8(self) -> u8 {
+            match self {
+                Mode::Reset => 0b10000000,
+                Mode::Mode(mode) => mode.to_u8(),
+            }
+        }
+    }
```

Comment:
I feel like using `serde` and treating this as a c-like enum would be a lot less code and much simpler to users.